dnf-behave-tests/features/steps/shell.py

```diff
@@ -42,12 +46,20 @@ def when_I_execute_in_shell(context, command):
     context.shell_session.sendline(command.format(context=context))
 
     if command.strip() == "quit" or command.strip() == "exit":
-        context.shell_session.expect(pexpect.EOF)
+        try:
+            context.shell_session.expect(pexpect.EOF)
+        except pexpect.exceptions.EOF:
+            stdout_from_shell(context)
+            raise AssertionError("dnf shell unexpected output")
```

Comment:
I don't think you'll get the `EOF` exception here, since you're expecting `EOF`. The other exception you can get is `TIMEOUT` (https://pexpect.readthedocs.io/en/stable/_modules/pexpect/exceptions.html).

The way I understand it you either get what you `expect()` or it waits for it until it times out. Might be worth it catching that too with some descriptive message like "expected prompt '> ' was not found in dnf shell output, timed out" (for the case where you expect the prompt)

Reply:
I see, yes you are right. Ok I have extracted it into a small wrapper for the exceptions and added more descriptive messages.